.github/workflows/workflow.yml

```diff
@@ -0,0 +1,44 @@
+name: workflow
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
```

Comment:
Read is default https://github.blog/changelog/2023-02-02-github-actions-updating-the-default-github_token-permissions-to-read-only/

🙃